Lib/test/test_codecs.py

```diff
@@ -3440,5 +3440,21 @@ def search_function(encoding):
         self.assertEqual(NOT_FOUND, codecs.lookup('a\xe9\u20ac-8'))
 
 
+class EncodingNormalizationTest(unittest.TestCase):
+
+    def test_normalization(self):
+        # encodings.normalize_encoding() ignores non-ASCII letters.
+        out = encodings.normalize_encoding('utf\xE9\u20AC\U0010ffff-8')
+        self.assertEqual(out, 'utf_8')
+        out = encodings.normalize_encoding('utf_8')
+        self.assertEqual(out, 'utf_8')
+        out = encodings.normalize_encoding('utf   8')
+        self.assertEqual(out, 'utf_8')
+        out = encodings.normalize_encoding('UTF 8')
```

Comment:
Would you mind to add a comment to explain that the function does not convert upper case letters to lower case letters, just to make the purpose of this test even more explicit?

Reply:
To be honest, I don't know how to explain it~ 
I found a case in https://github.com/python/cpython/blob/master/Lib/locale.py#L358. Looks like updating `encodings.normalize_encoding()` to conver to lower-case.

Reply:
just describe the fact, Lol~

Reply:
Maybe we can `enhance` this `encodings.normalize_encoding()`? I am not sure~